sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaWriteSchemaTransformProvider.java

```diff
@@ -131,10 +134,18 @@ public void finish() {
     @Override
     public PCollectionRowTuple expand(PCollectionRowTuple input) {
       Schema inputSchema = input.get("input").getSchema();
-      final SerializableFunction<Row, byte[]> toBytesFn =
-          configuration.getFormat().equals("JSON")
-              ? JsonUtils.getRowToJsonBytesFunction(inputSchema)
-              : AvroUtils.getRowToAvroBytesFunction(inputSchema);
+      final SerializableFunction<Row, byte[]> toBytesFn;
+      if (configuration.getFormat().equals("RAW")) {
+        if (!inputSchema.hasField(PAYLOAD)) {
```

Comment:
In PubSub, I've been just requiring a single payload (not caring about the name). Do we want to have the same pattern here?

Reply:
Yes, that's a tricky one. Kafka doesn't have the same attribute mechanism as PubSub. Maybe what we can do is allow the user to decide which "field" they want to pull from the Row. Then, we could have something like this:

```
    if len(field_names) != 1:
      raise ValueError(f'Expecting exactly one field, found {field_names}')
```
^ From the PubSub implementation. Then we do the input.getBytes(name_of_the_field_that_want_to_pull). I guess we need to come up with a good name for that like, getRowField. What do you think?

Reply:
If there's exactly one field we don't have to ask what it's name is, we can get it at pipeline construction time. (As another data point, this is what I did for write to text as well.)

Reply:
Yep, you're right. I pushed the necessary changes